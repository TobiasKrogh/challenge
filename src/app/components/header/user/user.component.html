<ng-container *ngIf="user">
  <em>
    {{ abbrevation }}
  </em>

  <span>
    {{ fullName }}
  </span>
</ng-container>
