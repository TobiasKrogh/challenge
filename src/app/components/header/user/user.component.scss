:host {
  --full-name-display: none;

  align-items: center;
  display: flex;

  @media (min-width: 480px) {
    --full-name-display: inline;
  }
}

em {
  align-items: center;
  background-color: var(--hex-color-yellow);
  border-radius: var(--border-radius--default);
  color: var(--hex-color-white);
  display: flex;
  font-size: 1.4375rem;
  font-style: normal;
  font-weight: var(--font-weight--bold);
  justify-content: center;

  height: 2.6875rem;
  width: 2.6875rem;
}

span {
  display: var(--full-name-display);
  margin-left: 0.6875rem;
}
