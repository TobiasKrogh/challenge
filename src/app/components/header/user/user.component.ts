import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { IUser } from '@challenge/api';

@Component({
  selector: 'challenge-user',
  templateUrl: './user.component.html',
  styleUrls: ['./user.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class UserComponent {
  @Input() public user: IUser | undefined;

  public get fullName(): string {
    return this.user ? `${this.user.firstName} ${this.user.lastName}` : '';
  }

  public get abbrevation(): string {
    return this.user
      ? `${this.user.firstName.charAt(0)}${this.user.lastName.charAt(0)}`
      : '';
  }
}
