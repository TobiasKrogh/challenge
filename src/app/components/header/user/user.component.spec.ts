import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MOCK_USER } from '../../../../testing/data.mocks';

import { UserComponent } from './user.component';

describe('UserComponent', () => {
  let component: UserComponent;
  let fixture: ComponentFixture<UserComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [UserComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(UserComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });

  describe('When rendering', () => {
    it('Then it matches snapshot (no user provided)', () => {
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });

    it('Then it matches snapshot (user provided)', () => {
      component.user = MOCK_USER;
      fixture.detectChanges();
      expect(fixture.debugElement).toMatchSnapshot();
    });
  });
});
