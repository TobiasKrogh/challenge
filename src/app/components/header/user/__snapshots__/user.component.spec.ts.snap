// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComponent When rendering Then it matches snapshot (no user provided) 1`] = `
DebugElement {
  "nativeNode": <div
    id="root1"
  >
    
  </div>,
}
`;

exports[`UserComponent When rendering Then it matches snapshot (user provided) 1`] = `
DebugElement {
  "nativeNode": <div
    id="root2"
  >
    <em>
       fl 
    </em>
    <span>
       firstName lastName 
    </span>
    
    
  </div>,
}
`;
